### DISCUSSION TITLE
Web3Modal asks the user to choose a wallet even though the wallet is connected on mobile

### DISCUSSION THREAD

**Discussion by Salman-Inayat** [WalletConnect/walletconnect-monorepo#2797] [Developer Support]
We are using WalletConnect, web3Modal, and Wagmi functions in our dapp. 
When a wallet is already connected, why does web3modal ask the user again to select a wallet for further actions, like depositing or withdrawing funds? It should go to the connected wallet and do the action. 

This is the configuration I am using. 

```
function App() {
  const chains = [mainnet];
  const projectId = process.env.REACT_APP_WALLETCONNECT_ID as string;
  
  const { publicClient } = configureChains(chains, [
    w3mProvider({ projectId }),
  ]);
  
  const wagmiConfig = createConfig({
    autoConnect: true,
    connectors: w3mConnectors({ projectId, version: 1, chains }),
    publicClient,
  });
  const ethereumClient = new EthereumClient(wagmiConfig, chains);
  
  const { isConnected } = useAccount();
  const { open, close } = useWeb3Modal();

....
```

```
return (
    <>
      <WagmiConfig config={wagmiConfig}>
         ...
      </WagmiConfig>
      <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
   </>
)
```

This is the deposit function that can be called if a wallet is already connected. 
```
const deposit = async()=> {
   const { request } = await prepareWriteContract({
      address: process.env.REACT_APP_CONTRACT as `0x${string}`,
      abi: abi,
      functionName: "DepositFunds",
      args: [],
      walletClient: props.ethereumClient,
      value: web3.utils.toWei(depositAmountInputValue),
  });

 const { hash } = await writeContract(request);
}
```

In the above function, the writeContract() method is causing the web3modal to pop up and ask the user to select a wallet, but it should directly go to the connected wallet. 

Any help would be really appreciated.

**Comment by glitch-txs** [comment]
Hi, could you share a minimal reproducible example repo of the issue?

**Comment by xzilja** [comment]
First things first, wagmi config needs to be taken outside of your render loop (outside of App component), let's see if that fixes the issue

**Reply by Salman-Inayat** [reply]
Hi @0xAsimetriq 
I moved wagmi config out of the App component but still getting a popup to select the wallet even when a wallet is connected. I want to bypass the OS-specific(android in this case) popup to select the wallet. 

![photo_2023-06-28 13 39 34](https://github.com/WalletConnect/walletconnect-monorepo/assets/45660519/edc83566-3d74-45e4-88c9-2e6ce40d5d0c)

**Reply by Salman-Inayat** [reply]
@0xAsimetriq 
Can you help me with this?

**Reply by xzilja** [reply]
This pops up when a connection or other requests are made i.e. we redirect user to a deeplink like `wc://` when action needs to be performed in a wallet. Android's native behavior is to present this modal with options of apps that can handle the deeplink (native behavior is out of our control).

You can delete `WALLETCONNECT_DEEPLINK_CHOICE` localstorage item to prevent redirects, but this could lead to unexpected behavior and users not knowing they need to continue in  a wallet.

**Reply by Salman-Inayat** [reply]
I delete the `WALLETCONNECT_DEEPLINK_CHOICE` from the local storage but the popup didn't open, though I received a notification from the other wallet that some action is required. 
Seems like I have to go with the default behavior of selecting wallet every time there is some activity.

**Reply by xzilja** [reply]
Some android versions also show things like "Always" or "Just Once" where you can select wallet that will open by default without needing to show this prompt. I assume most users will use that option, but as devs we have multiple wallets we test against, so showing that selection is somewhat required.

**Comment by Salman-Inayat** [comment]
Is there a way to get the connected wallet info somehow using walletconnect and then use that wallet in the deeplink?

**Reply by xzilja** [reply]
From `WALLETCONNECT_DEEPLINK_CHOICE`, however all wallets on android share same deeplink `wc://`
